### PR TITLE
Dgc 2106 python support version

### PR DESCRIPTION
**Pull request by Jerry-Jinfeng-Guo**

### CODE REVIEW

File: docs/examples/pandapower_example.ipynb [hunk 1]
```diff
@@ -84,7 +84,16 @@
...
     "scrolled": true
    },
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "c:\\Users\\AL29065\\AppData\\Local\\miniconda3\\envs\\power-grid-io-env\\lib\\site-packages\\pandapower\\create.py:4815: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'Dyn' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
+      "  net[element_type].at[index, column] = value\n"
```
Comment (mgovers) [inline docs/examples/pandapower_example.ipynb:93]: please create a new issue for this. either we need to fix it if it is in PGM-IO, else if it is in pandapower, we should suppress this warning

File: docs/examples/pandapower_example.ipynb [hunk 2]
```diff
@@ -317,14 +348,14 @@
...
+   "execution_count": 14,
    "metadata": {},
    "outputs": [],
    "source": [
     "from power_grid_model_io.converters import PgmJsonConverter\n",
     "    \n",
-    "input_file = \"data/pandapower/example_simple_input.json\"\n",
-    "output_file = \"data/pandapower/example_simple_output.json\"\n",
+    "input_file = \"data/pandapower/example_simple_input_test.json\"\n",
+    "output_file = \"data/pandapower/example_simple_output_test.json\"\n",
```
Comment (mgovers) [inline docs/examples/pandapower_example.ipynb:358]: where does this change come from?

File: src/power_grid_model_io/converters/pandapower_converter.py [hunk 3]
```diff
@@ -2389,4 +2389,4 @@ def lookup_id(self, pgm_id: int) -> Dict[str, Union[str, int]]:
                 if name:
                     return {"table": table, "name": name, "index": indices[pgm_id]}
                 return {"table": table, "index": indices[pgm_id]}
-        raise KeyError(pgm_id)
+        raise KeyError(pgm_id)
```
Comment (mgovers) [inline src/power_grid_model_io/converters/pandapower_converter.py:2392]: newline
```suggestion
        raise KeyError(pgm_id)

```

Comment (Jerry-Jinfeng-Guo) [reply src/power_grid_model_io/converters/pandapower_converter.py]: Resolved in revert

File: docs/examples/pandapower_example.ipynb (hunk 1 above)
Comment (Jerry-Jinfeng-Guo) [reply docs/examples/pandapower_example.ipynb]: Discussed during stand-up

**Review by mgovers — APPROVED** [review]

**Pull request merged**